scripts/normalize_user_tags.py

```diff
@@ -136,20 +142,24 @@ def add_prereg_campaign_tags():
     """
     Tag = apps.get_model('osf', 'Tag')
     OSFUser = apps.get_model('osf', 'OSFuser')
-    prereg_challenge_source_tag = Tag.all_tags.get(name=CampaignSourceTags.PreregChallenge.value, system=True)
-    prereg_source_tag = Tag.all_tags.create(name=CampaignSourceTags.Prereg.value, system=True)
-    logger.info('Added tag ' + prereg_source_tag.name)
-    prereg_challenge_cutoff_date = pytz.utc.localize(datetime(2019, 01, 01, 05, 59))
-    prereg_users_registered_after_january_first = OSFUser.objects.filter(tags__id=prereg_challenge_source_tag.id, date_registered__gt=prereg_challenge_cutoff_date)
-    logger.info('Number of OSFUsers created on/after 2019-01-01: ' + str(len(prereg_users_registered_after_january_first)))
-    for user in prereg_users_registered_after_january_first:
-        user.tags.through.objects.filter(tag=prereg_challenge_source_tag, osfuser=user).delete()
-        user.add_system_tag(prereg_source_tag)
-        # Check  whether the user is merged to another user
-        # If so, add the same tag to that user as well
-        if user.merged_by is not None:
-            user.merged_by.add_system_tag(prereg_source_tag)
-    logger.info('Migrated users from ' + prereg_challenge_source_tag.name + ' to ' + prereg_source_tag.name)
+    try:
+        prereg_challenge_source_tag = Tag.all_tags.get(name=CampaignSourceTags.PreregChallenge.value, system=True)
+        prereg_source_tag, created = Tag.all_tags.get_or_create(name=CampaignSourceTags.Prereg.value, system=True)
+        logger.info('Added tag ' + prereg_source_tag.name)
+        prereg_challenge_cutoff_date = pytz.utc.localize(datetime(2019, 01, 01, 05, 59))
+        prereg_users_registered_after_january_first = OSFUser.objects.filter(tags__id=prereg_challenge_source_tag.id, date_registered__gt=prereg_challenge_cutoff_date)
+        logger.info('Number of OSFUsers created on/after 2019-01-01: ' + str(len(prereg_users_registered_after_january_first)))
+        for user in prereg_users_registered_after_january_first:
+            user.tags.through.objects.filter(tag=prereg_challenge_source_tag, osfuser=user).delete()
+            user.add_system_tag(prereg_source_tag)
+            # Check  whether the user is merged to another user
+            # If so, add the same tag to that user as well
+            if user.merged_by is not None:
+                user.merged_by.add_system_tag(prereg_source_tag)
+        logger.info('Migrated users from ' + prereg_challenge_source_tag.name + ' to ' + prereg_source_tag.name)
+    except Tag.DoesNotExist:
+        prereg_source_tag, created = Tag.all_tags.get_or_create(name=CampaignSourceTags.Prereg.value, system=True)
+        logger.info('Added tag ' + prereg_source_tag.name)
```

Comment:
You should try to keep a minimum amount of code in your try blocks, you shoul just use it to potentially catch the error that getting a nonexistant `prereg_challenge_source` tag would cause.  I'd move most of this code out of the try/except.